### PR TITLE
Buildah java tests

### PR DESCRIPTION
**Pull request by theexplorist**
Kindly review the tests added @dev-gaur @rohanKanojia .

### CODE REVIEW

**Comment by rohanKanojia** [comment]
@theexplorist : cool, thanks. I'll review them as soon as I get time.

File: buildah-core/src/test/java/io/jshift/buildah/core/commands/CommandTransformer.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package io.jshift.buildah.core.commands;
+
+public class CommandTransformer {
+
+    public static String[] transform(String command) {
```
Comment (rohanKanojia) [inline buildah-core/src/test/java/io/jshift/buildah/core/commands/CommandTransformer.java:5]: Why do you need to create a new class just for String split operation?

File: pom.xml [hunk 2]
```diff
@@ -97,10 +99,20 @@
         <artifactId>org.eclipse.jgit</artifactId>
         <version>${jgit.version}</version>
       </dependency>
+      <dependency>
+        <groupId>org.awaitility</groupId>
```
Comment (rohanKanojia) [inline pom.xml]: Where is this dependency used?

File: pom.xml [hunk 3]
```diff
@@ -97,10 +99,20 @@
...
+        <version>${awaitility.version}</version>
+      </dependency>
       <dependency>
         <groupId>org.zeroturnaround</groupId>
         <artifactId>zt-exec</artifactId>
-        <version>1.11</version>
+        <version>${zt-exec.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.yaml</groupId>
```
Comment (rohanKanojia) [inline pom.xml]: Same here, I could not see any places where you used any class of this package. Is this really required?

File: buildah-core/src/test/java/io/jshift/buildah/core/resolvers/UrlLocationResolverTest.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    private boolean isUpAndRunning(String url) throws IOException {
+        final URL request = new URL(url);
+
+        HttpURLConnection requestConnection =  ( HttpURLConnection )  request.openConnection ();
+        requestConnection.setRequestMethod ("HEAD");
+        requestConnection.connect () ;
+        int code = requestConnection.getResponseCode() ;
+        requestConnection.disconnect();
+
+        return code > 199 && code < 300;
```
Comment (rohanKanojia) [inline buildah-core/src/test/java/io/jshift/buildah/core/resolvers/UrlLocationResolverTest.java]: maybe we can avoid using hardcoded statuses here? How about using `HttpStatus` ?

**Review by rohanKanojia — COMMENTED** [review]
Overall looks good to me, but I see there are lots of code copied from the odo-java . Could we maybe refactor to avoid code duplication? If not, maybe add a reference ?

**Comment by theexplorist** [comment]
> Overall looks good to me, but I see there are lots of code copied from the odo-java . Could we maybe refactor to avoid code duplication? If not, maybe add a reference ?

@rohanKanojia actually resolvers are working the same as in odo java so that part is taken from there, rest I have added test for each command here. And tested them all. Any suggestions from your side or any changes apart from the unused dependencies, please let me know!

**Review by rohanKanojia — APPROVED** [review]

**Pull request merged**